site/src/pages/ManagementSettingsPage/IdpSyncPage/ExportPolicyButton.stories.tsx

```diff
@@ -0,0 +1,69 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { expect, fn, userEvent, waitFor, within } from "@storybook/test";
+import {
+	MockGroupSyncSettings,
+	MockOrganization,
+	MockRoleSyncSettings,
+} from "testHelpers/entities";
+import { ExportPolicyButton } from "./ExportPolicyButton";
+
+const meta: Meta<typeof ExportPolicyButton> = {
+	title: "modules/resources/ExportPolicyButton",
+	component: ExportPolicyButton,
+	args: {
+		policy: JSON.stringify(MockGroupSyncSettings, null, 2),
+		type: "groups",
+		organization: MockOrganization,
+	},
+};
+
+export default meta;
+type Story = StoryObj<typeof ExportPolicyButton>;
+
+export const Default: Story = {};
+
+export const ClickExportGroupPolicy: Story = {
+	args: {
+		policy: JSON.stringify(MockGroupSyncSettings, null, 2),
+		type: "groups",
+		organization: MockOrganization,
+		download: fn(),
+	},
+	play: async ({ canvasElement, args }) => {
+		const canvas = within(canvasElement);
+		await userEvent.click(
+			canvas.getByRole("button", { name: "Export Policy" }),
+		);
+		await waitFor(() =>
+			expect(args.download).toHaveBeenCalledWith(
+				expect.anything(),
+				`${MockOrganization.name}_groups-policy.json`,
+			),
+		);
+		const blob: Blob = (args.download as jest.Mock).mock.calls[0][0];
```

Comment:
oh this is so gross (but I don't know what to suggest to make it better)

could you add a comment above it to elaborate why this incantation is justified/necessary? I _think_ I understand what it's doing, but clarity is good, and more junior devs might not get it!